src/webviews/extension-side/dataviewer/dataViewerDependencyService.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { SemVer } from 'semver';
+import { ProductNames } from '../../../kernels/installer/productNames';
+import { Product } from '../../../kernels/installer/types';
+import { IApplicationShell } from '../../../platform/common/application/types';
+import { traceWarning } from '../../../platform/logging';
+import { IsCodeSpace } from '../../../platform/common/types';
+import { parseSemVer } from '../../../platform/common/utils';
+import { DataScience, Common } from '../../../platform/common/utils/localize';
+import { EnvironmentType, PythonEnvironment } from '../../../platform/pythonEnvironments/info';
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import { IDataViewerDependencyService } from './types';
+import { executeSilently } from '../../../kernels/helpers';
+import { IJupyterSession, IKernel, IKernelProvider } from '../../../kernels/types';
+
+const minimumSupportedPandaVersion = '0.20.0';
+
+function isVersionOfPandasupported(version: SemVer) {
+    return version.compare(minimumSupportedPandaVersion) > 0;
+}
+
+/**
+ * Responsible for managing dependencies of a Data Viewer.
+ */
+@injectable()
+export class DataViewerDependencyService implements IDataViewerDependencyService {
+    private get kernel(): IKernel {
+        const kernel = this.kernelProvider.kernels.find((kernel) => kernel.status === 'idle');
```

Comment:
This looks wrong, we're just looking for any random kernel, this method could even end up returning an idle `rust` kernel

Reply:
Whoops I missed this. You have an interpreter (which by the way isn't going to work in the web version). That could be used to find a matching kernel.

However at the point where this service is called, you could just pass it the kernel instead. Don't send it the interpreter, send it the kernel in use.

Reply:
That was my original suggestion to Daniel, to pass the kernel into this method

Reply:
I have resolved this! Thank you for pointing this out.